src/ORM/Association.php

```diff
@@ -438,6 +438,16 @@ public function property($name = null)
         if ($name === null && !$this->_propertyName) {
             list(, $name) = pluginSplit($this->_name);
             $this->_propertyName = Inflector::underscore($name);
+            $columns = array_keys($this->_sourceTable->schema()->columns());
+            if (in_array($this->_propertyName, $columns)) {
+                $msg = 'Association property name "%s" clashes with field of same name of table "%s".'
+                    . "\n"
+                    . 'You should explicitly specify the "propertyName" option.';
+                trigger_error(
```

Comment:
Throwing can exception should not be done in a bugfix release imo as someone could be relying on this behavior. We can change to exception for 3.3